SocialCareCaseViewerApi.Tests/V1/UseCase/Residents/GetAllUseCaseTests.cs

```diff
@@ -0,0 +1,185 @@
+using AutoFixture;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SocialCareCaseViewerApi.Tests.V1.Helpers;
+using SocialCareCaseViewerApi.V1.Boundary.Requests;
+using SocialCareCaseViewerApi.V1.Boundary.Response;
+using SocialCareCaseViewerApi.V1.Gateways;
+using SocialCareCaseViewerApi.V1.Infrastructure;
+using SocialCareCaseViewerApi.V1.UseCase;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SocialCareCaseViewerApi.Tests.V1.UseCase.Residents
+{
+    [TestFixture]
+    public class GetAllUseCaseTests
+    {
+        private GetAllUseCase _getAllUseCase;
+        private Mock<IDatabaseGateway> _mockDatabaseGateway;
+        private Fixture _fixture = new Fixture();
+
+        [SetUp]
+        public void SetUp()
+        {
+            _mockDatabaseGateway = new Mock<IDatabaseGateway>();
+            _getAllUseCase = new GetAllUseCase(_mockDatabaseGateway.Object);
+
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(
+                 It.IsAny<int>(),
+                 It.IsAny<int>(),
+                 It.IsAny<long>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>(),
+                 It.IsAny<string>())
+            ).Returns(new List<ResidentInformation>());
+        }
+
+        [Test]
+        public void ReturnsEmptyListWhenMosaicIdIsProvidedInLettersOnlyFormat()
+        {
+            var request = new ResidentQueryParam() { MosaicId = "nonNumeric" };
+
+            _getAllUseCase.Execute(request, cursor: 0, limit: 20)
+                .Should()
+                .BeEquivalentTo(new ResidentInformationList() { Residents = new List<ResidentInformation>() });
+        }
+
+        [Test]
+        public void IfLimitLessThanTheMinimumWillUseTheMinimumLimit()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(0, 10, null, null, null, null, null, null, null))
+                .Returns(new List<ResidentInformation>()).Verifiable();
+
+            _getAllUseCase.Execute(new ResidentQueryParam(), 0, 4);
+
+            _mockDatabaseGateway.Verify();
+        }
+
+        [Test]
+        public void IfLimitMoreThanTheMaximumWillUseTheMaximumLimit()
+        {
+            _mockDatabaseGateway.Setup(x => x.GetResidentsBySearchCriteria(0, 100, null, null, null, null, null, null, null))
+                .Returns(new List<ResidentInformation>()).Verifiable();
+
+            _getAllUseCase.Execute(new ResidentQueryParam(), 0, 400);
+
+            _mockDatabaseGateway.Verify();
+        }
```

Comment:
I think we could improve the readability and understanding of these tests as I struggled at first. For example, seeing the lone `_mockDatabaseGateway.Verify();` as the expectation was initially confusing. I eventually got it and understood that we're expecting that the database gateway to be called with a limit of 10/100 but because we've stubbed the gateway method to only reply to those arguments so we can just use `Verify()`.

Maybe we can remove the stubbing here as we're already doing it in the `SetUp` and update the expectation `Verify()` to check the arguments to be 10/100?

I also think it would help to extract variables for these numbers e.g. `var maximumLimit = 100` so it's clearer.

Reply:
Possibly silly question but what happens on the boundary when the limit is 10? 😅

Reply:
I have updated the setup and used constants for the limits. Hopefully it's more clear now. I've also added tests for the boundary values to make it clear what happens in those situations.